dev/NumberBox/NumberBox.cpp

```diff
@@ -0,0 +1,344 @@
+{
+    m_formatter.IsZeroSigned(IsZeroSigned());
+}
+
+void NumberBox::OnRoundingAlgorithmPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateRounder();
+}
+void NumberBox::OnNumberRounderPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateRounder();
+}
+void NumberBox::OnIncrementPrecisionPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateRounder();
+}
+void NumberBox::OnSignificantDigitPrecisionPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateRounder();
+}
+
+void NumberBox::OnValuePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    if (m_TextBox)
+    {
+       // UpdateTextToValue();
+    }
+}
+
+
+
+// Trigger any validation, rounding, and processing done onLostFocus
+void NumberBox::OnTextBoxLostFocus(winrt::IInspectable const& sender, winrt::RoutedEventArgs const& args)
+{
+    ValidateInput();
+}
+
+// Performs all validation steps on input given in textbox. Runs on LoseFocus and stepping.
+void NumberBox::ValidateInput()
+{
+    std::string InputAsString = winrt::to_string(m_TextBox.Text());
+    // Handles Empty TextBox Case, current behavior is to set Value to default (0)
+    if (InputAsString == "")
+    {
+        Value(0);
+        m_hasError = false;
+        return;
+    }
+
+    winrt::IReference<double> parsedNum = m_formatter.ParseDouble(m_TextBox.Text());
+    // TODO: Formatter.Format(value)
+
+    if (parsedNum && IsInBounds(parsedNum.Value()) )
+    {
+        SetErrorState(false);
+        Value(parsedNum.Value());
+        UpdateTextToValue();
+    }
+    else
+    {
+        SetErrorState(true);
+    }
+}
+
+// SpinClicks call to decrement or increment, 
+void NumberBox::OnSpinDownClick(winrt::IInspectable const&  sender, winrt::RoutedEventArgs const& args)
+{
+    StepValue(false);
+}
+void NumberBox::OnSpinUpClick(winrt::IInspectable const& sender, winrt::RoutedEventArgs const& args)
+{
+    StepValue(true);
+}
+void NumberBox::OnNumberBoxKeyUp(winrt::IInspectable const& sender, winrt::KeyRoutedEventArgs const& args)
+{
+    switch (args.Key()) {
+        case winrt::VirtualKey::Up:
+        case winrt::VirtualKey::GamepadDPadUp:
+            StepValue(true);
+            break;
+        case winrt::VirtualKey::Down:
+        case winrt::VirtualKey::GamepadDPadDown:
+            StepValue(false);
+            break;
+    }
+}
+void NumberBox::OnScroll(winrt::IInspectable const& sender, winrt::PointerRoutedEventArgs const& args) {
+    if (!HyperScrollEnabled()) {
+        return;
+    }
+    int delta = args.GetCurrentPoint(*this).Properties().MouseWheelDelta();
+
+    if (delta > 0)
+    {
+        StepValue(true);
+    }
+    else if (delta < 0)
+    {
+        StepValue(false);
+    }
+}
+
+// Increments or decrements value by StepFrequency, wrapping if necessary
+void NumberBox::StepValue(bool sign)
```

Comment:
>sign [](start = 31, length = 4)

maybe isPositive is a better name

Reply:
Enums are also nice as they make it clear what the intent of the passed parameter value is.